### DISCUSSION TITLE
Cannot figure out how to use plugins

### DISCUSSION THREAD

**Discussion by leo-petrucci** [ionic-team/capacitor#3128] [Q&A] [answered]
Hi all, super new to this, so sorry if it's something really easy.

I understand the basics of Capacitor, I've been able to export a React app to a Native app without many problems.

I now want to use some location services, I've decided to go with [this plugin](https://github.com/triniwiz/capacitor-fancy-geo) for no particular reason. Now the plugin itself doesn't really explain what I'm supposed to do to use it. If I try to use the code it's telling me in the repository I just get a bunch of `undefined` errors.

So I go to the Capacitor docs for "[Using plugins](https://capacitor.ionicframework.com/docs/basics/using-plugins)". This page then links me to "[Plugins](https://capacitor.ionicframework.com/docs/plugins/)" page which in return explains how to create a plugin from scratch, which isn't really what I'm looking for.

How do I use plugins? Is there a part of the docs I've missed? Do I need to add support for Typescript to use plugins? How do I use Cordova plugins?

**Comment by dotNetkow** [comment]
No apology needed - we'll add more details to that Using Plugins page shortly. In the meantime - 

0. If you haven't already, add a platform like 'npx cap add android'.

1. Install the plugin: npm i capacitor-fancy-geo
2. Sync the project. This ensures the native bits are installed into the android/ios project and my guess as to why you see undefined errors.   `npx cap sync`.  If that's successful, you'll see a message like "Found 1 Capacitor plugin for android: capacitor-fancy-geo (0.0.1)"
3. Use the plugin as detailed [here](https://github.com/triniwiz/capacitor-fancy-geo#usage).

Try those steps - does that work?

**Comment by leo-petrucci** [comment]
@dotNetkow Thank you for replying! Doing this outputs the following error when trying to run the app via Android Studio:

```
Could not find any matches for ionic-team:capacitor-android:1+ as no versions of ionic-team:capacitor-android are available.
Searched in the following locations:
  - https://dl.google.com/dl/android/maven2/ionic-team/capacitor-android/maven-metadata.xml
  - https://dl.google.com/dl/android/maven2/ionic-team/capacitor-android/
  - https://jcenter.bintray.com/ionic-team/capacitor-android/maven-metadata.xml
  - https://jcenter.bintray.com/ionic-team/capacitor-android/
  - file:/home/leonardo/Desktop/Node Projects/time-management/android/capacitor-cordova-android-plugins/src/main/libs/
  - file:/home/leonardo/Desktop/Node Projects/time-management/android/app/libs/
Required by:
    project :app > project :capacitor-fancy-geo
```

The app runs fine after removing the plugin. Unsure if that's the plugin's fault or I'm missing an npm package.

**Reply by leo-petrucci** [reply]
Also having trouble simply creating a plugin, but that might be because I'm not super familiar with Java. Is there any information if plugins work without publishing first? Can I just run `npm run build` within my plugin project and then locally install it with `npm install "path/to/my/plugin"?

The `.java` file is located at `src/main/java/com/example/location/LocationPlugin.java` but I can't seem to import `import com.example.app.LocationPlugin;`. All Android Studio reports is:

```
error: cannot find symbol
import com.example.app.LocationPlugin;
```

**Reply by leo-petrucci** [reply]
I think the problem is related to `npx cap sync` not working correctly. If I understand it correctly, this is supposed to copy the plugin's java files into my app. But having looked everywhere in my project the files just haven't been copied at all.

The commands runs:
```
✔ Copying web assets from build to android/app/src/main/assets/public in 26.30ms
✔ Copying native bridge in 4.04ms
✔ Copying capacitor.config.json in 1.58ms
✔ copy in 44.72ms
✔ Updating Android plugins in 1.79ms
  Found 1 Capacitor plugin for android:
    my-plugin (0.0.1)
✔ update android in 11.68ms
✔ copy in 254.64μp
✔ update web in 5.81μp
Sync finished in 0.064s
```

But I'm not sure if it's actually working the way it's supposed to be. I also posted a detailed walkthrough of my process over on the [Ionic Forums](https://forum.ionicframework.com/t/developing-plugins-cant-impor-to-mainactivity-java/191548) for a more detailed overview of what's happening.

EDIT: I can confirm the plugins aren't being copied. I manually wrote the same plugin and placed it at `android/app/src/main/java/com/example/myplugin/MyPlugin.java` and I was able to import it into `MainActivity.java` by just doing `import com.example.myplugin.MyPlugin;`.

I tested it in the app itself and it works there just fine returning the value I need.

**Comment by leo-petrucci** [comment, marked as answer]
Okay so I think I've managed to fix it. I completely uninstalled Android Studio and reinstalled it from snap. I'm unsure how useful this was to fixing it but I wanted to be sure nothing had gone wrong with the installation.

The problem (I think) was caused by my Gradle Sync not working. It wasn't triggering automatically and even when manually started it wouldn't complete (even though it returned no errors). 

After the install I still need to manually Sync to have new plugins appear, but aside from that it's now working just fine!

**Comment by amitkhare** [comment]
Update 
````
....
versionCode 
versionName 
....
````
at build.gradle 
to a higher number then sync.



worked for me.
